mathesar/api/serializers/records.py

```diff
@@ -48,6 +48,10 @@ def update(self, instance, validated_data):
                     table=table,
                     status_code=status.HTTP_400_BAD_REQUEST,
                 )
+            elif "CheckViolation" in str(type(e.orig)):
```

Comment:
It would be better to compare with the actual exception type rather than the string of the exception class. Something like `type(e.orig) == CheckViolation`